src/test/java/org/jabref/gui/externalfiles/FileFilterUtilsTest.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    @Test
+    public void isDuringLastWeekNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(7)), false);
+    }
+
+    @Test
+    public void isDuringLastWeekPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(6).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastMonthNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(30)), false);
+    }
+
+    @Test
+    public void isDuringLastMonthPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(29).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastYearNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(365)), false);
+    }
+
+    @Test
+    public void isDuringLastYearPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(364).minusHours(23)), true);
+    }
+
+    @Nested
+    class SortingTests {
+
+        private List<Path> files = new ArrayList<Path>();
+        private List<String> expectedSortByDateAscending = new ArrayList<String>();
+        private List<String> expectedSortByDateDescending = new ArrayList<String>();
+        private List<String> wrongOrder = new ArrayList<String>();
+
+        /* Initialize the directory and files used in the sorting tests, and change their last edited dates. */
+        @BeforeEach
+        public void setUp() throws Exception {
+            String dirPath = "src/test/resources/unlinkedFiles";
+
+            Path path = Path.of(dirPath);
+            Files.createDirectory(path);
+    
+            Path firstPath = Path.of(dirPath.concat("/firstFile.pdf"));
+            Path secondPath = Path.of(dirPath.concat("/secondFile.pdf"));
+            Path thirdPath = Path.of(dirPath.concat("/thirdFile.pdf"));
+            Path fourthPath = Path.of(dirPath.concat("/fourthFile.pdf"));
+    
+            Files.createFile(firstPath);
+            Files.createFile(secondPath);
+            Files.createFile(thirdPath);
+            Files.createFile(fourthPath);
+
+            // change the files last edited times.
+            File firstFile = new File(firstPath.toString());
+            File secondFile = new File(secondPath.toString());
+            File thirdFile = new File(thirdPath.toString());
+            File fourthFile = new File(fourthPath.toString());
+    
+            ZoneId zoneId = ZoneId.systemDefault();
+    
+            LocalDateTime firstDate = LocalDateTime.of(2021, 1, 10, 0, 0, 0, 0);
+            LocalDateTime secondDate = LocalDateTime.of(2021, 1, 5, 0, 0, 0, 0);
+            LocalDateTime thirdDate = LocalDateTime.of(2021, 1, 1, 0, 0, 0, 0);
+            LocalDateTime fourthDate = LocalDateTime.of(2021, 1, 2, 0, 0, 0, 0);
+    
+            firstFile.setLastModified(firstDate.atZone(zoneId).toEpochSecond());
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#setLastModifiedTime-java.nio.file.Path-java.nio.file.attribute.FileTime-

Reply:
> LGTM as well! You can ignore the mac, the action doesn't work on forks because it requires access to some secrets (and that are not avaiable on forks)

Thank you very much, so the main thing left now is to deal with the NPE @calixtus  mentioned. We can either handle it separately or include the code in the above try catch block  so that the NPE will never be triggered